docs/user-guide/framework/bootstrap-components/infrastructure.md

```diff
@@ -23,4 +27,131 @@ Ignition can bootstrap the following services for the Infrastructure API:
+
+        :param str template: template of infrastructure to be created
+        :param str template_type: type of template used i.e. TOSCA or Heat
+        :param str inputs: values for the inputs defined on the tosca template
+        :param dict deployment_location: the deployment location to deploy to
+        :return: an ignition.model.infrastructure.CreateInfrastructureResponse
+        """
+        pass
+
+    @interface
+    def get_infrastructure_task(self, infrastructure_id, request_id, deployment_location):
+        """
+        Get information about the infrastructure (created or deleted)
+
+        :param str infrastructure_id: identifier of the infrastructure to check
+        :param str request_id: identifier of the request to check
+        :param dict deployment_location: the location the infrastructure was deployed to
+        :return: an ignition.model.infrastructure.InfrastructureTask instance describing the status
+        """
+        pass
+
+    @interface
+    def delete_infrastructure(self, infrastructure_id, deployment_location):
+        """
+        Initiates a request to delete infrastructure previously created with the given infrastructure_id.
+        This method should return immediate response of the request being accepted,
+        it is expected that the InfrastructureService will poll get_infrastructure_task on this driver to determine when the request has complete.
+
+        :param str infrastructure_id: identifier of the infrastructure to be removed
+        :param dict deployment_location: the location the infrastructure was deployed to
+        :return: an ignition.model.infrastructure.DeleteInfrastructureResponse
+        """
+        pass
+
+    @interface
+    def find_infrastructure(self, template, template_type, instance_name, deployment_location):
+        """
+        Finds infrastructure instances that meet the requirements set out in the given TOSCA template, returning the desired output values from those instances
+
+        :param str template: tosca template of infrastructure to be found
+        :param str template_type: type of template used i.e. TOSCA or Heat
+        :param str instance_name: name given as search criteria
+        :param dict deployment_location: the deployment location to deploy to
+        :return: an ignition.model.infrastructure.FindInfrastructureResponse
+        """
+        pass
+```
+
+There are several methods required on any Service implementing this Capability, in order to serve Create/Delete/Find infrastructure requests. The following sections describe the flow of an incoming request, outlines the role of Ignition and how your driver implementation is used.
+
+## On Create
+
+The InfrastructureApiService handles the incoming HTTP request, parsing the headers and body before making a call to the InfrastructureService with the expected arguments. 
```

Comment:
Probably a pain to do (if even possible), but is it worth adding a sequence diagram here?

Reply:
I agree, just ran out of time really to add it. I'll raise that as a separate issue (https://github.com/accanto-systems/ignition/issues/33) so we can do it when time allows.